File: Modul 2/second-task.py
import operator

print('\n<<<Kegiatan 2>>>\n')

firstList = ['jurusan', 'praktikum', 'kampus', 'tahun']
secondList = ['informatika', 'fungsional', 'UMM', '2020']

def zippedWords(firstWords, secondWords):
    words = []
    for (firstWord, secondWord) in zip(firstWords, secondWords):
        words.append(operator.add(firstWord, secondWord))
    return words

def convertWords(words, letterCase):
    return map(lambda word: defineLetterCase(word, letterCase), words)

def defineLetterCase(word, letterCase):
    if letterCase == 'lowercase':
        return word.lower()
    elif letterCase == 'uppercase':
        return word.upper()
    elif letterCase == 'capitalize':
        return word.capitalize()
    else:
        return word

print(zippedWords(firstList, secondList))

zippedSecondList = zippedWords(firstList, secondList)
print(list(convertWords(zippedSecondList, 'lowercase')))

converFirstList = convertWords(firstList, 'capitalize')
convertSecondList = convertWords(secondList, 'capitalize')
print(tuple(zippedWords(converFirstList, convertSecondList)))

zippedFourthList = zippedWords(firstList, secondList)
print(tuple(convertWords(zippedFourthList, 'uppercase')))

zippedDict = zip(firstList, secondList)
print(dict(zippedDict))

#Robb - Mobile & Web Enthusiast

File: Modul 2/second-task.js
const firstList = ['jurusan', 'praktikum', 'kampus', 'tahun'];
const secondList = ['informatika', 'fungsional', 'UMM', '2020'];

function combineElement(firstList, secondList) {
    const newList = [];
    for (let firstItem = 0; firstItem < firstList.length; firstItem++) {
        for (let secondItem = 0; secondItem < secondList.length; secondItem++) {
            if (firstItem === secondItem) {
                newList.push(firstList[firstItem] + secondList[secondItem])
            }
        }
    }
    return newList
}

function convertLetterCase(wordList, letterCase) {
    return wordList.map((wordItem) => defineLetterCase(wordItem, letterCase));
}

function defineLetterCase(wordItem, letterCase) {
    if (letterCase === 'uppercase') {
        return wordItem.toUpperCase();
    } else if (letterCase === 'lowercase') {
        return wordItem.toLowerCase();
    } else {
        return wordItem;
    }
}

console.log(combineElement(firstList, secondList))
